htdocs/product/class/product.class.php

```diff
@@ -1167,14 +1168,34 @@ public function update($id, $user, $notrigger = false, $action = 'update', $upda
 					// Quantities in batch details are not same as stock quantity,
 					// so we add a default batch record to complete and get same qty in parent and child table
 					if ($ObjW->real <> $qty_batch) {
-						$ObjBatch = new Productbatch($this->db);
-						$ObjBatch->batch = $valueforundefinedlot;
-						$ObjBatch->qty = ($ObjW->real - $qty_batch);
-						$ObjBatch->fk_product_stock = $ObjW->id;
-
-						if ($ObjBatch->create($user, 1) < 0) {
+						if ($this->status_batch == 2) {
```

Comment:
With old code, it was possible to fix a corrupted qty for product with serial. So after it was possible to record a stock correction to get a qty of 1.
If we block here, it means we can't solve a pb of corrupted data for such product ?